File: java-code/src/runnables/WriteFile.java
package runnables;

import classes.Manager;
import classes.base.Employee;

import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;
import java.util.List;

public class WriteFile implements Runnable {

  private final String _content;
  private final String _fileName;

  public WriteFile(List<Employee> list, String fileName) {
    this._fileName = fileName;
    // String builder is used to be a string concatenation device better than a `+=`
    // Uses less memory as `+=` makes a new whole copy each time - so large lists can
    // take up a lot of memory
    StringBuilder sb = new StringBuilder();
    for (Employee item : list) {
      sb.append(item.GetInformation()).append("\n");
    }
    _content = sb.toString();
  }

  @Override
  public void run(){
    try (BufferedWriter writer = new BufferedWriter(new FileWriter(_fileName))) {
      System.out.println("Writing content to " + this._fileName);
      writer.write(_content);
    } catch (IOException e) {
      System.err.println("Error writing to file: " + e);
    }
  }
}

File: c#-code/src/runnables/WriteFile.cs
namespace APLCAssignment3.Runnables
{
    using APLCAssignment3.Classes;
    using APLCAssignment3.Classes.Base;  
    using System.Collections.Generic;
    using System.IO;
    using System;
    using System.Text;

    public class WriteFile
    {
        private readonly string _content;
        private readonly string _fileName;

        public WriteFile(List<Employee> list, string fileName)
        {
            this._fileName = fileName;
            
            StringBuilder sb = new StringBuilder(); 
            
            foreach (Employee item in list) 
            {
                sb.Append(item.GetInformation()).Append(Environment.NewLine);
            }
            
            _content = sb.ToString();
        }

        public void Run()
        {
            try
            {
                Console.WriteLine($"Writing content to {this._fileName}");
            
                File.WriteAllText(_fileName, _content);

            }
            catch (IOException e)
            {
                Console.Error.WriteLine($"Error writing to file {e.Message}");
            }
        }
    }
}